code/datums/gamemodes/game_mode.dm

```diff
@@ -92,6 +92,17 @@
 		var/datum/DBQuery/query_round_game_mode = SSdbcore.NewQuery("UPDATE [format_table_name("round")] SET [sql] WHERE id = [GLOB.round_id]")
 		query_round_game_mode.Execute()
 		qdel(query_round_game_mode)
+	spawn_roundstart_weapons()
+
+/datum/game_mode/proc/spawn_roundstart_weapons()
+	while(GLOB.weapon_spawn_list.len)
```

Comment:
dont use this style of loop unless you really really really need to milk out performance